src/linter/block.go

```diff
@@ -1176,10 +1183,31 @@ func (b *BlockWalker) handleNew(e *expr.New) bool {
 		return true
 	}
 
-	if _, ok := meta.Info.GetClass(className); !ok {
+	info, ok := meta.Info.GetClass(className)
+	if !ok {
 		b.r.Report(e.Class, LevelError, "undefined", "Class not found %s", className)
 	}
 
+	// Check implicitly invoked constructor method arguments count.
+	ctor, ok := info.Methods["__construct"]
```

Comment:
Constructor can be defined in children so that's why I use solver.FindMethod everywhere. Constructors can also be private (or protected) so it be also worth trying to check the context as well.